packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -167,17 +201,32 @@ export class SelectedNetworkController extends BaseController<
 
     this.update((state) => {
       state.domains[domain] = networkClientId;
-      if (!state.perDomainNetwork) {
-        state.domains[METAMASK_DOMAIN] = networkClientId;
-      }
     });
   }
 
+  domainHasPermissions(domain: Domain): boolean {
+    return this.messagingSystem.call(
+      'PermissionController:hasPermissions',
+      domain,
+    );
+  }
+
   setNetworkClientIdForDomain(
     domain: Domain,
     networkClientId: NetworkClientId,
   ) {
-    if (!this.state.perDomainNetwork) {
+    // Early return if perDomainNetwork is disabled and the domain is not Metamask, unless it's setting for Metamask specifically
+    if (domain !== METAMASK_DOMAIN && !this.state.perDomainNetwork) {
+      return;
+    }
+
+    // Check and, if not a metamask request, return early if the domain lacks permissions
+    if (domain !== METAMASK_DOMAIN && !this.domainHasPermissions(domain)) {
+      return;
+    }
+
+    // If setting for Metamask and perDomainNetwork is disabled, update all domains to the Metamask networkClientId
```

Comment:
this ensures that the proxies stay in sync

Reply:
i think it would be helpful if the comment mentioned keeping proxies in sync